src/object_schema.cpp

```diff
@@ -330,6 +343,9 @@ void ObjectSchema::validate(Schema const& schema, std::vector<ObjectSchemaValida
         validate_property(schema, name, prop, &primary, exceptions);
     }
 
+    if (!primary_key.empty() && is_embedded) {
+        exceptions.emplace_back("Embedded table '%1' cannot have primary key '%2'.", name, primary_key);
```

Comment:
Just to make it more clear that no primary key is allowed at all.
```suggestion
        exceptions.emplace_back("Embedded table '%1' is not allowed to define a primary key. Found key '%2'.", name, primary_key);
```